app/models.py

```diff
@@ -33,6 +71,33 @@ def by_user(db, user: FlathubUser):
     def by_gh_id(db, ghid):
         return db.session.query(GithubAccount).filter_by(github_userid=ghid).first()
 
+    @staticmethod
+    def delete_hash(hasher: utils.Hasher, db, user):
+        account = GithubAccount.by_user(db, user)
+        hasher.add_number(account.github_userid)
+        hasher.add_string(account.login)
+        repos = [repo.reponame for repo in GithubRepository.all_by_account(db, account)]
+        repos.sort()
+        for repo in repos:
+            hasher.add_string(repo)
+
+    @staticmethod
+    def delete_user(db, user):
+        gha = GithubAccount.by_user(db, user)
+        db.session.execute(
+            delete(GithubRepository).where(GithubRepository.github_account == gha.id)
+        )
+        db.session.delete(gha)
+        # Clear the user's details
+        user.display_name = None
+        # Mark the user as deleted
+        user.deleted = True
+        # Ensure the DB is updated
+        db.session.add(user)
+        db.session.commit()
```

Comment:
I think all this belongs in the user deletion function, not here in githubaccount.